Remove fade from footer display transition

peppy opened the pull request:
Fades should not be used for these kinds of elements. The opacity changes of multiple elements looks shocking. It's unnecessary.

Before:

https://github.com/user-attachments/assets/943a189d-a1e5-414e-9190-223f77965525

After:

https://github.com/user-attachments/assets/35a373cd-5368-4ecb-9b6d-63273010b9ea

peppy committed:
Fix incorrectly named variable

peppy commented on osu.Game/Screens/Footer/ScreenFooter.cs:
```diff
@@ -61,7 +61,10 @@ public partial class ScreenFooter : OverlayContainer
-            Height = HEIGHT;
+
+            // For masking and transition purposes, take up the full height of displayed buttons.
+            Height = ScreenFooterButton.HEIGHT;
```
fuuuuuuuuuuuuuuu

peppy commented on osu.Game/Screens/Footer/ScreenFooter.cs:
```diff
@@ -78,13 +81,23 @@ private void load()
-                    RelativeSizeAxes = Axes.Both,
+                    Anchor = Anchor.BottomLeft,
+                    Origin = Anchor.BottomLeft,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
```
I think I've fixed this.

bdach replied:
If you mean https://github.com/ppy/osu/pull/33664/commits/23653b722b4e77837a86a9da954379979b70dd0a then I'm not sure that did anything at all...

peppy replied:
I think I posted that on the wrong PR somehow.

bdach commented:
Really sorry to keep doing this but mod select overlay looks broke now:

<img width="1624" alt="Screenshot 2025-06-19 at 16 49 16" src="https://github.com/user-attachments/assets/04854ada-40a0-4ca6-ac34-c8b1f697329d" />

Something to do with messing with the relative size / autosize thing, no doubt. I'm not making any judgement call on the sanity of that code on master either, though.

peppy committed:
Fix tests

peppy committed:
Remove fade from footer display transition

Fades should not be used for these kinds of elements. The opacity
changes of multiple elements looks shocking. It's unnecessary.

peppy commented:
I've rewritten this to not touch anything that breaks things because I can't make it work. See what you think. The new approach just disables masking rather than work around it.

bdach commented on osu.Game/Screens/Footer/ScreenFooter.cs:
```diff
@@ -40,7 +40,11 @@ public partial class ScreenFooter : OverlayContainer
+
+        // Disable masking because it breaks due to the height of this container being less than the displayed content.
+        // The height being set as it is is required for transition purposes.
+        public override bool UpdateSubTreeMasking() => false;
```
I'm not entirely sure what this is supposed to be fixing (as in I don't see anything break when this is commented out), but I suppose I'll just smile and nod to finally move this forward?
